PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/runner/deploy/SpringCloudDeploymentState.java

```diff
@@ -175,142 +157,40 @@ protected String getDeployTarget() {
+            DefaultLoader.getUIHelper().showErrorNotification(NOTIFICATION_TITLE, "Failed to get application url");
+        } else {
+            setText(processHandler, String.format("Application url: %s", publicUrl));
         }
     }
 
-    private String getDependenciesValidationPrompt(List<String> missingDependencies,
-                                                   Map<String, String> inCompatibleDependencies, String springVersion) {
-        StringBuilder result = new StringBuilder();
-        result.append(DEPENDENCY_WARNING);
-        for (String dependency : missingDependencies) {
-            result.append(String.format("%s : Missing \n", dependency));
-        }
-        for (String dependency : inCompatibleDependencies.keySet()) {
-            result.append(String.format("%s : Incompatible, current version %s, spring boot version %s \n",
-                                        dependency, inCompatibleDependencies.get(dependency), springVersion));
-        }
-        return result.toString();
+    @SneakyThrows
+    private static AppResourceInner getInner(final SpringCloudAppEntity app) {
+        final Field inner = SpringCloudAppEntity.class.getDeclaredField("inner");
+        inner.setAccessible(true);
+        return (AppResourceInner) inner.get(app);
     }
 
-    private Map<String, String> getSpringAppDependencies(Enumeration<JarEntry> jarEntryEnumeration,
-                                                         String libraryPath) {
-        final String[] springArtifacts = ArrayUtils.add(SPRING_ARTIFACTS, SPRING_BOOT_AUTOCONFIGURE);
-        final List<JarEntry> jarEntries = Collections.list(jarEntryEnumeration);
-        return jarEntries.stream()
-                         .filter(jarEntry -> StringUtils.startsWith(jarEntry.getName(), libraryPath)
-                                 && StringUtils.equalsIgnoreCase(FilenameUtils.getExtension(jarEntry.getName()), JAR))
-                         .map(jarEntry -> {
-                             String fileName = FilenameUtils.getBaseName(jarEntry.getName());
-                             final int i = StringUtils.lastIndexOf(fileName, "-");
-                             return (i > 0 && i < fileName.length() - 1) ?
-                                    new String[]{
-                                            StringUtils.substring(fileName, 0, i),
-                                            StringUtils.substring(fileName, i + 1)
-                                    } :
-                                    new String[]{fileName, ""};
-                         })
-                         .filter(entry -> ArrayUtils.contains(springArtifacts, entry[0]))
-                         .collect(Collectors.toMap(entry -> entry[0], entry -> entry[1]));
+    @SneakyThrows
+    private static DeploymentResourceInner getInner(final SpringCloudDeploymentEntity deployment) {
+        final Field inner = SpringCloudDeploymentEntity.class.getDeclaredField("inner");
```

Comment:
why we use reflection to get inner?

Reply:
add todo

Reply:
https://dev.azure.com/mseng/VSJava/_workitems/edit/1812809